### PR TITLE
[HTML] Introducing tag_indents & tag_styles

### PR DESCRIPTION
**Pull request by Lucas-C**
This is a follow-up of #1066 that also provides the feature requested in #1074

**Checklist**:

- [x] The GitHub pipeline is OK (green), meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

### Styling HTML tags globally

The style of several HTML tags (`<a>`, `<blockquote>`, `<code>`, `<pre>`, `<li>`, `<h1>`, `<h2>`, `<h3>`...) can be set globally, for the whole HTML document, by passing `tag_styles` to `FPDF.write_html()`:

```python
from fpdf import FPDF, FontFace

pdf = FPDF()
pdf.add_page()
pdf.write_html("""
  <h1>Big title</h1>
  <section>
    <h2>Section title</h2>
    <p>Hello world!</p>
  </section>
""", tag_styles={
    "h1": FontFace(color=(148, 139, 139), size_pt=32),
    "h2": FontFace(color=(148, 139, 139), size_pt=24),
})
pdf.output("html_styled.pdf")
```

Similarly, the indentation of several HTML tags (`<blockquote>`, `<dd>`, `<li>`) can be set globally, for the whole HTML document, by passing `tag_indents` to `FPDF.write_html()`:

```python
from fpdf import FPDF

pdf = FPDF()
pdf.add_page()
pdf.write_html("""
  <dl>
      <dt>Term</dt>
      <dd>Definition</dd>
  </dl>
""", tag_indents={"dd": 5})
pdf.output("html_dd_indented.pdf")
```

### CODE REVIEW

**Comment by Lucas-C** [comment]
@gmischler & @andersonhc: could you please review this PR?

It introduces some new "structuring" parameters to `HTML2FPDF`, so I'd be happy to get your feedback on that 😊

File: fpdf/html.py [hunk 1]
```diff
@@ -825,6 +977,30 @@ def error(self, message):
         raise RuntimeError(message)
 
 
+def ul_prefix(ul_type):
```
Comment (gmischler) [inline fpdf/html.py:980]: No CoerciveEnums for those two?

**Review by gmischler — APPROVED** [review]
Looks ok, adding flexibility without making them much more complicated.

File: fpdf/html.py (hunk 1 above)
Comment (Lucas-C) [reply fpdf/html.py:980]: The special case with `if len(ul_type) == 1` makes it difficult to convert this to a `CoerciveEnum`.

But if's deemed preferable, we could still perform this conversion in the future. It's not part of the user-facing functionalities.

**Comment by Lucas-C** [comment]
> Looks ok, adding flexibility without making them much more complicated.

Thank you for the feedback!

Merging this now.

**Pull request merged**